src/include/migraphx/literal.hpp

```diff
@@ -98,10 +98,18 @@ struct literal : raw_data<literal>
     std::vector<literal> get_sub_objects() const { return {}; }
 
     /// Convert the data to an argument
-    argument get_argument() const
+    argument get_argument(bool weight_streaming = false) const
```

Comment:
There is no need to do this. You can create an operator that stores an `argument` and just returns that:

```cpp
struct literal_as_argument
{
    std::string name() const { return "gpu::literal_as_argument"; }
    argument data;

    template <class Self, class F>
    static auto reflect(Self& self, F f)
    {
        return pack(f(self.data, "data"));
    }

    shape compute_shape(const std::vector<shape>&) const
    {
        return data.get_shape();
    }
    argument compute_shape(const std::vector<argument>&) const
    {
        return data;
    }
};
MIGRAPHX_REGISTER_OP(literal_as_argument);
```

And then replace the literals with this operator in the write_literals pass.

Reply:
Should this operator be created in a new header file or in the same file as `literal.hpp`?

Reply:
I would create it directly in the write_literals.cpp file.

Reply:
I tried putting it directly into the `write_literals.cpp` file but it's not as drop and replace as I expected. During execution it comes up with a `/code/AMDMIGraphX/src/include/migraphx/operation.hpp:359: compute_op: Not computable: gpu::literal_as_argument` error, any idea what this means? Not too familiar with how operators work.

Reply:
Yea its missing the `compute` method. I mistyped the `compute` function. The second `compute_shape` is supposed to be `compute`.

Reply:
Hmm that doesn't seem to work either (same error), I have the following as my operator:
```cpp
struct literal_as_argument
{
    std::string name() const { return "gpu::literal_as_argument"; }
    argument data;
    std::string data_name = "{ ... }";

    template <class Self, class F>
    static auto reflect(Self& self, F f)
    {
        return pack(f(self.data_name, "data"));
    }

    shape compute_shape(const std::vector<shape>&) const
    {
        return data.get_shape();
    }
    argument compute(const std::vector<argument>&) const
    {
        return data;
    }
};
MIGRAPHX_REGISTER_OP(literal_as_argument);
```

I changed the reflect function to avoid having the entire data field print out every time, but that shouldn't be affecting the operation.

Reply:
Fixed and pushed, thanks